feat(recordings): filter by events in session recording

alexkim205 opened the pull request:
## Problem

Global playlist filters should identify matching events in session recording lists.

<!-- Who are we building for, what are their needs, why is this important? -->

## Changes

From Insight

https://user-images.githubusercontent.com/13460330/192576836-7d4c00ae-6773-4975-aa02-3e5cee5af279.mov

From Recording Playlist

https://user-images.githubusercontent.com/13460330/192576854-96e3fda3-9e41-43ec-98a1-67fd1bb112b2.mov


<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

TODO in a separate PR: Fuzzy matching

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Logic tests for new filtering logic.

alexkim205 committed:
move eventsData to eventslistlogic

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/matching-events

alexkim205 committed:
refactor openSessionPlayeR

alexkim205 committed:
refactor activeSessionRecordingId

alexkim205 committed:
matching

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/matching-events

alexkim205 committed:
matching works

alexkim205 committed:
rename id to uuid

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/matching-events

alexkim205 commented on posthog/queries/session_recordings/session_recording_list.py:
```diff
@@ -355,7 +355,7 @@ def _data_to_return(self, results: List[Any]) -> List[Dict[str, Any]]:
                 "matching_events": [
                     {
                         "events": [
-                            dict(zip(["timestamp", "id", "session_id", "window_id"], event)) for event in row[i + 1]
```
renaming to uuid to remain consistent with matching events data returned from persons query

alexkim205 committed:
fix and add a bunch of tests

alexkim205 committed:
add eventslist tests

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/matching-events

alexkim205 committed:
fix tests

alexkim205 committed:
Merge remote-tracking branch 'origin' into feat/matching-events

alexkim205 committed:
start with default new entity filter

rcmarron commented on frontend/src/scenes/session-recordings/player/list/sharedListLogic.ts:
```diff
@@ -30,6 +31,12 @@ export const sharedListLogic = kea<sharedListLogicType>([
                 setTab: (_, { tab }) => tab,
             },
         ],
+        showOnlyMatching: [
```
I think we should move this to the `playerSettingsLogic` or somewhere that isn't keyed on the recording. That way, you can toggle this setting and then quickly click through recordings in the playlist, and the setting is persisted

alexkim205 replied:
good idea!

rcmarron commented on frontend/src/types.ts:
```diff
@@ -2050,6 +2055,7 @@ export interface SessionRecordingPlayerProps {
     playerKey: string
     includeMeta?: boolean
     recordingStartTime?: string
+    matching?: MatchedRecording[]
```
I think this name is a bit vague. Should it be `matchingEvents` and then the type be `MatchedRecordingEvents[]`?

rcmarron replied:
I wonder if it should be a dict then? Like `matching.events = []`. If we do want to expand the prop to other types, then we probably don't want those other types interlaced in the same lis.t

rcmarron commented on frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.test.ts:
```diff
@@ -0,0 +1,109 @@
+import recordingMetaJson from 'scenes/session-recordings/__mocks__/recording_meta.json'
+import recordingEventsJson from 'scenes/session-recordings/__mocks__/recording_events.json'
+
+describe('sessionRecordingPlayerLogic', () => {
```
❤️

rcmarron reviewed (commented):
Looking great! A few nits in the comments. Capturing the couple things we just talked about too here, so we don't miss them:

* Info bubble on filter control
* Make the matching event pins colored and move to forefront
* List filter should have an empty default event
* Maybe some weird bug with matching events being empty

alexkim205 commented:
Addressed all four points of these points! + added fuzzy matching 

> * Info bubble on filter control
> * Make the matching event pins colored and move to forefront
> * List filter should have an empty default event
> * Maybe some weird bug with matching events being empty


https://user-images.githubusercontent.com/13460330/192844785-9a8023ba-ce50-4f92-b973-4adf38936d15.mov

alexkim205 committed:
addressed feeddback

alexkim205 committed:
connect
